x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx

```diff
@@ -0,0 +1,318 @@
+          newPackagePolicy || packagePolicy,
+          packageInfo,
+          safeLoad
+        );
+        setValidationResults(newValidationResult);
+        // eslint-disable-next-line no-console
+        console.debug('Package policy validation results', newValidationResult);
+
+        return newValidationResult;
+      }
+    },
+    [packagePolicy, packageInfo]
+  );
+  // Update package policy method
+  const updatePackagePolicy = useCallback(
+    (updatedFields: Partial<NewPackagePolicy>) => {
+      const newPackagePolicy = {
+        ...packagePolicy,
+        ...updatedFields,
+      };
+      setPackagePolicy(newPackagePolicy);
+
+      // eslint-disable-next-line no-console
+      console.debug('Package policy updated', newPackagePolicy);
+      const newValidationResults = updatePackagePolicyValidation(newPackagePolicy);
+      const hasPackage = newPackagePolicy.package;
+      const hasValidationErrors = newValidationResults
+        ? validationHasErrors(newValidationResults)
+        : false;
+      const hasAgentPolicy = newPackagePolicy.policy_id && newPackagePolicy.policy_id !== '';
+      if (
+        hasPackage &&
+        (hasAgentPolicy || selectedPolicyTab === SelectedPolicyTab.NEW) &&
+        !hasValidationErrors
+      ) {
+        setFormState('VALID');
+      } else {
+        setFormState('INVALID');
+      }
+    },
+    [packagePolicy, setFormState, updatePackagePolicyValidation, selectedPolicyTab]
+  );
+
+  const onSaveNavigate = useOnSaveNavigate({
+    packagePolicy,
+    queryParamsPolicyId,
+  });
+
+  const navigateAddAgent = (policy?: PackagePolicy) =>
+    onSaveNavigate(policy, ['openEnrollmentFlyout']);
+
+  const navigateAddAgentHelp = (policy?: PackagePolicy) =>
+    onSaveNavigate(policy, ['showAddAgentHelp']);
+
+  const onSubmit = useCallback(
+    async ({
+      force,
+      overrideCreatedAgentPolicy,
+    }: { overrideCreatedAgentPolicy?: AgentPolicy; force?: boolean } = {}) => {
+      if (formState === 'VALID' && hasErrors) {
+        setFormState('INVALID');
+        return;
+      }
+      if (agentCount !== 0 && formState !== 'CONFIRM') {
+        setFormState('CONFIRM');
+        return;
+      }
+      let createdPolicy = overrideCreatedAgentPolicy;
+      if (selectedPolicyTab === SelectedPolicyTab.NEW && !overrideCreatedAgentPolicy) {
+        try {
+          setFormState('LOADING');
+          if ((withSysMonitoring || newAgentPolicy.monitoring_enabled?.length) ?? 0 > 0) {
+            const packagesToPreinstall: string[] = [];
+            if (packageInfo) {
+              packagesToPreinstall.push(packageInfo.name);
+            }
+            if (withSysMonitoring) {
+              packagesToPreinstall.push(FLEET_SYSTEM_PACKAGE);
+            }
+            if (newAgentPolicy.monitoring_enabled?.length ?? 0 > 0) {
+              packagesToPreinstall.push(FLEET_ELASTIC_AGENT_PACKAGE);
+            }
+
+            if (packagesToPreinstall.length > 0) {
```

Comment:
If we are adding system or elastic-agent package policy, the same package won't be installed twice, right?

Reply:
I am going to check just to be sure